src/compiler/checker.ts

```diff
@@ -10145,7 +10145,29 @@ namespace ts {
                                     error(accessExpression, Diagnostics.Element_implicitly_has_an_any_type_because_type_0_has_no_index_signature_Did_you_mean_to_call_1, typeToString(objectType), suggestion);
                                 }
                                 else {
-                                    error(accessExpression, Diagnostics.Element_implicitly_has_an_any_type_because_type_0_has_no_index_signature, typeToString(objectType));
+                                    let errorInfo: DiagnosticMessageChain | undefined;
+                                    if (indexType.flags & TypeFlags.EnumLiteral) {
+                                        errorInfo = chainDiagnosticMessages(/* details */ undefined, Diagnostics.Property_0_does_not_exist_on_type_1, "[" + typeToString(indexType) + "]", typeToString(objectType));

```

Comment:
can these just be converted into `createDiagnostic` or `error` or something instead? `chainDiagnosticMessage` is more for concatenation.

Reply:
`reportError` shells out to `chainDiagnosticMessages` internally within relationship checking and builds a stack, but this is within `getPropertyTypeForIndexType`, so `chainDiagnosticMessages` is the only way to build an error stack. `error` does return its initial diagnostic, however, so using `chainDiagnosticMessages` with an `undefined` details isn't strictly required.

Reply:
@weswigham but `error` returns an `Diagnostic` so I can't chain anything to it using `chainDiagnosticMessages` a bit lower in the file (also `error` already adds the message to the diagnostics collection, which again I don't want since this is just part of the error).

Or am I misunderstanding the comment here, and the suggestion is to not use chained diagnostics at all and just report a simple one line error. I deliberately chose the chaining approach as I felt the information here would have been a bit too long, and it felt right to first specify that the type of the whole index expression is incompatible with the object and the delve deeper as to the root cause of the incompatibility.

Reply:
@weswigham Hi did you get a change to look at the above comment? Do you have any feedback for me ? I would like to have this ready and this is the only outstanding review comment.  👀

Reply:
Oh, my comment was mostly to @DanielRosenwasser to remark that explicit chaining is probably the best way to make a mulitiline message here, since our other tools for doing so are limited to specific scopes.

Reply:
@weswigham Oh, my bad then 😊. Guess if there is nothing else then this is pretty much ready to merge whenever you guys are ready for it.